### DISCUSSION TITLE
What OS on Rpi5?

### DISCUSSION THREAD

**Discussion by mntbighker** [iRobotEducation/create3_docs#613] [Questions]
### How are you connecting to your Create 3?

Wi-Fi and USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

MacOS / Rpi5 Ubuntu

### Which version of ROS 2 is installed on your computer?

Other

### Which firmware version is installed on your robot?

I.0.0.FastDDS (Iron)

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

Home

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

As of Sept. 2024 what is the preferred OS to run on a USB connected Pi5? It seems Ubuntu was until recently the only option. But it seems ROS2 supports Bookworm now. Does the maturity of Ubuntu ROS make it the better choice? Or would Bookwork make for a more efficient setup. I feel like Ubuntu is a "bulkier" choice if it can be avoided. I started with Ubuntu and no Docker. It appears ROS2, and hopefully Fast DDS will run native on Bookworm without Docker. I'm fishing for opinions, and my Create3 just arrived yesterday. My main use for the bot will be to annoy the cat, so I don't need to ;-)  I hope to mess with LIDAR at some point, but I don't want to spend thousands on an Nvidia box. I was hoping to possibly install the Rpi AI hat.

**Comment by slowrunner** [comment]
Won’t yo have to compile EVERYTHING? ROS 2, nav2, slam_toolbox, and the LIDAR driver, behavior trees, and fastdds.

PiOS has the best Pi5 support including the RTC, vcgencmd, sleep and raspi-config, but Ubuntu has the best ROS 2 support with lots of folks with similar configuration.

I brought up Ubuntu 24.04 (desktop), installed Open-SSH, and ROS 2 Jazzy with no issues in less than 30 minutes.  (I spent two days trying to get a VNC server configured and failed because my desktop is a Mac.  Have been told RealVNC will work, or if your client is Windows Remote Desktop U24.04 includes Remote Desktop support built-in).

Once you are developing your code, everything is identical PiOS or Ubuntu.

I run Humble in Docker on my Pi5 to keep the Pi5 PiOS features available and can tell you Docker has complicated every step forward.  I hate it, and can’t wait till I can go native with Jazzy (and a working Remote Desktop Solution.  There is very little need for the Remote Desktop, it makes looking at maps and image captures easier).

As for the 8TOPS Hailo AI hat - I suggest you settle for a 4TOPS AI processor with stereo depth cameras and RGB camera and IMU in an OAK-D camera.  I started with the Oak-D-Lite, and have graduated to the OAK-D-W97 to have wide angle and depth for every pixel.  The 4TOPS NN processor has enough oomph to do 30fps color point clouds and YOLO object reco with x,y,z.

Also powering the Pi5 with the AI hat is much more complicated than powering just the Pi5 off the USB-C and powering the camera and LIDAR off a buck from the Create3 14v.
